### PR TITLE
Proposal: Green Lemon Protocol🍋 - An anonymous NFT solution

### PR DESCRIPTION
**Pull request by wuyahuang**
### Project Abstract

[Proposal: Dual-Key Stealth Address Protocol](https://github.com/w3f/Grants-Program/pull/997)

[Milestone Delivery: Dual-Key Stealth Address Protocol](https://github.com/w3f/Grant-Milestone-Delivery/pull/504)

The Green Lemon protocol is an anonymous NFT solution for the Polkadot ecosystem based on zero-knowledge proof and dual-key stealth address protocol: users deposit DOT to an anonymous NFT contract and then anonymously send mint, transfer, and other ERC721 functions to that contract via relayer.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying?

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] I have read and understood the [FAQs](https://github.com/w3f/Grants-Program/blob/master/docs/faq.md), [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md) and [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/announcement-guidelines.md).
- [x] A BTC, Ethereum (USDT/USDC/DAI) or Polkadot/Kusama (aUSD) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledge the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### How Did You Hear About our grants program?

- [ ] Social Media
- [ ] Hackathon
- [ ] Personal Recommendation
- [ ] Substrate Builders Program
- [ ] Investor/VC
- [x] Online Search
- [ ] Other: _______

### CODE REVIEW

**Review by Noc2 — APPROVED** [review]
Thanks for the application and sorry for the late response here. I was waiting for the approval of your delivery and forgot to comment here. I’m generally happy to support the project and share it with the rest of the team. But maybe you also want to integrate an article into the deliverables (see[ https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--implement-substrate-modules) )

**Comment by wuyahuang** [comment]
Hi @Noc2, glad to see you again. I have added the `article` module into the deliverables. Thank you.

**Review by Noc2 — APPROVED** [review]

**Review by alxs — APPROVED** [review]

**Review by laboon — APPROVED** [review]

**Pull request merged**
